pkg/apis/extensions/types.go

```diff
@@ -982,6 +993,19 @@ const (
 	RunAsUserStrategyRunAsAny RunAsUserStrategy = "RunAsAny"
 )
 
+// RunAsGroupStrategy denotes strategy types for generating RunAsGroup values for a
+// SecurityContext.
+type RunAsGroupStrategy string
+
+const (
+	// container must run as a particular gid.
+	RunAsGroupStrategyMustRunAs RunAsGroupStrategy = "MustRunAs"
+	// container must run as a non-root gid
+	RunAsGroupStrategyMustRunAsNonRoot RunAsGroupStrategy = "MustRunAsNonRoot"
+	// container may make requests for any gid.
+	RunAsGroupStrategyRunAsAny RunAsGroupStrategy = "RunAsAny"
```

Comment:
Other potentially useful strategies: [MayRunAs](https://github.com/kubernetes/kubernetes/issues/56173), "MatchUser" (i.e. must be unset or set to the same as the RunAsUser).